src/main/java/mate/academy/springboot/datajpa/repository/specification/requestparam/GetAllProductByPriceRange.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.springboot.datajpa.repository.specification.requestparam;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.specification.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class GetAllProductByPriceRange implements SpecificationProvider<Product> {
```

Comment:
I don't think criteria query is needed for this request. This is a regular request that can be used using the JpaRepository method of the interface

Reply:
what is the difference between my current implementation and your proposal?